src/pk-main.c

```diff
@@ -173,10 +173,14 @@ main (int argc, char *argv[])
 	/* get values from the config file */
 	conf = g_key_file_new ();
 	conf_filename = pk_util_get_config_filename ();
+	if (conf_filename == NULL) {
+		g_error ("Cannot found config file.");
```

Comment:
g_error is fatal and won't do any of the cleanup. It just needs to be a `g_print` and then `goto out` kind of cleanup.